flujo/processors/diff.py

```
@@ -0,0 +1,98 @@
+
+    @staticmethod
+    def _extract_pair(data: object) -> tuple[object, object]:
+        if isinstance(data, dict) and "before" in data and "after" in data:
+            return data["before"], data["after"]
+        if isinstance(data, (list, tuple)) and len(data) == 2:
+            return data[0], data[1]
+        before = getattr(data, "before", None)
+        after = getattr(data, "after", None)
+        if before is not None or after is not None:
+            return before, after
+        raise ValueError("DiffProcessor expects {'before':..., 'after':...} or (before, after)")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)